pkg/kubelet/dockershim/docker_image.go

```diff
@@ -98,10 +98,15 @@ func (ds *dockerService) RemoveImage(image *runtimeapi.ImageSpec) error {
 			}
 		}
 		return nil
+	} else if err != nil && dockertools.IsImageNotFoundError(err) {
+		return nil
 	}
 
 	_, err = ds.client.RemoveImage(image.Image, dockertypes.ImageRemoveOptions{PruneChildren: true})
-	return err
+	if err != nil && !isImageNotFoundError(err) {
```

Comment:
Actually, why can't we modify `kubeDockerClient.RemoveImage` to return the correct `ImageNotFoundError`?